docker-compose.yml

```diff
@@ -503,18 +498,48 @@ services:
   # Utility Services
   ########################################################################
 
+  grapl-pulumi:
+    image: grapl/grapl-local-pulumi:${TAG:-latest}
+    command: |
+      /bin/bash -c "
+         wait-for-it ${S3_HOST}:${S3_PORT_INTERNAL} --timeout=60 &&
+         wait-for-it ${LOCALSTACK_HOST}:${LOCALSTACK_PORT} --timeout=60 &&
+         sleep 10 &&
+         pulumi login --local &&
+         pulumi stack init local-grapl --non-interactive &&
+         # See pulumi/README.md fro why we do these configuration overrides
+         pulumi config set aws:accessKey ${FAKE_AWS_ACCESS_KEY_ID} --plaintext &&
+         pulumi config set aws:secretKey ${FAKE_AWS_SECRET_ACCESS_KEY} --plaintext &&
+         pulumi config set --path 'aws:endpoints[0].s3' ${S3_ENDPOINT} &&
+         pulumi up --yes --skip-preview --stack=local-grapl &&
+         # NOTE: This is running in localstack's network namespace, so
+         # we will need to wait on localstack, NOT grapl-pulumi.
+         python -u -m http.server ${WAIT_PORT}
```

Comment:
Can you tell me a bit more about how this universal WAIT_PORT works? Is it just, every container exposes :1337 and that's just the magic "wait for it" port? No conflict over the docker network? 

(If so: wonderful)

Reply:
Yup, basically. Just like you can have multiple servers on the same network that expose a service on the same port, you can have multiple containers do the same thing because they each have their own network namespace by default.

(In this particular instance, however, the `grapl-pulumi` container happens to intentionally share the network namespace of the `localstack` container, which is why we wait on `localstack:${WAIT_PORT}`.)

I basically created the `WAIT_PORT` variable so we didn't have to manually track lots of different magic port numbers.